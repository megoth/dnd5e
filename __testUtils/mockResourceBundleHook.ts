import mockSWR from "./mockSWR";
import mockResourceBundle from "./mockResourceBundle";

export default function mockResourceBundleHook(
  mock,
  bundle = mockResourceBundle()
) {
  return mockSWR(mock, { data: bundle });
}
