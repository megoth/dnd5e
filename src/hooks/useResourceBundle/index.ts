import { useContext } from "react";
import { ResourceBundleContext } from "../../contexts/resourceBundle";

export default function useResourceBundle() {
  return useContext(ResourceBundleContext);
}
