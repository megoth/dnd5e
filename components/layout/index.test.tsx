import React from "react";
import { render } from "@testing-library/react";
import Layout from "./index";
import useResourceBundle from "../../src/hooks/useResourceBundle";
import mockResourceBundleHook from "../../__testUtils/mockResourceBundleHook";

jest.mock("../../src/hooks/useResourceBundle");
const mockedResourceBundleHook = useResourceBundle as jest.Mock;

describe("Layout", () => {
  beforeEach(() => mockResourceBundleHook(mockedResourceBundleHook));

  it("renders as non-home by default", () => {
    const { asFragment } = render(<Layout>test</Layout>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("can be rendered as home", () => {
    const { asFragment } = render(<Layout home>test</Layout>);
    expect(asFragment()).toMatchSnapshot();
  });
});
