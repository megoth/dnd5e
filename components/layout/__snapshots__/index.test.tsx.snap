// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout can be rendered as home 1`] = `
<DocumentFragment>
  <div
    class="flex-1 flex flex-col relative min-h-full"
  >
    <header
      class="mb-2 shadow shadow-md"
    >
      <div
        class="main-container main-container--content"
      >
        <div
          class="font-serif self-center text-left"
        >
          <a
            class="focus:outline-none focus:ring-2 focus:ring-red-600"
            href="/"
          >
            [Translation for https://example.com/global/translations.ttl#appName does not exist]
          </a>
        </div>
      </div>
      <nav
        class="pages-nav pages-nav--main-menu"
      >
        <ul
          class="pages-nav__list pages-nav__list--main-menu"
        >
          <li
            class="flex"
            data-testid="main-nav-list-item"
          >
            <a
              class="pages-nav__link pages-nav__link--main-menu"
              href="/characters"
            >
              [Translation for https://example.com/global/translations.ttl#charactersPageTitle does not exist]
            </a>
          </li>
          <li
            class="flex"
            data-testid="main-nav-list-item"
          >
            <a
              class="pages-nav__link pages-nav__link--main-menu pages-nav__link--selected pages-nav__link--child-selected"
              href="/rules"
            >
              [Translation for https://example.com/global/translations.ttl#rulesPageTitle does not exist]
            </a>
            <button
              class="pages-nav__button pages-nav__button--main-menu lg:hidden"
              data-testid="nav-item-sub-menu-button"
              type="button"
            >
              <svg
                class="icon"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 6h16M4 12h16M4 18h16"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </li>
        </ul>
      </nav>
    </header>
    <main>
      test
    </main>
    <footer
      class="main-container main-container--content mt-8 text-center sm:text-left xs:flex xs:space-x-4 items-center justify-center sm:justify-start"
    >
      <div
        class="md:flex-grow pb-2 flex space-x-1 justify-center sm:justify-start"
      >
        <a
          class="link"
          href="/about"
        >
          [Translation for https://example.com/global/translations.ttl#aboutTitle does not exist]
        </a>
        <a
          class="link"
          href="/faq"
        >
          [Translation for https://example.com/global/translations.ttl#faqShort does not exist]
        </a>
      </div>
      <div
        class="pb-2 flex xs:space-x-2 space-y-2 xs:space-y-0 flex-col text-center xs:flex-row xs:justify-center"
      >
        <button
          class="button"
          data-testid="dark-mode-selector-button"
          type="button"
        >
          <span>
            ðŸŒ™Â [Translation for https://example.com/global/translations.ttl#darkModeTurnOn does not exist]
          </span>
        </button>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`Layout renders as non-home by default 1`] = `
<DocumentFragment>
  <div
    class="flex-1 flex flex-col relative min-h-full"
  >
    <header
      class="mb-2 shadow shadow-md"
    >
      <div
        class="main-container main-container--content"
      >
        <div
          class="font-serif self-center text-left"
        >
          <a
            class="focus:outline-none focus:ring-2 focus:ring-red-600"
            href="/"
          >
            [Translation for https://example.com/global/translations.ttl#appName does not exist]
          </a>
        </div>
      </div>
      <nav
        class="pages-nav pages-nav--main-menu"
      >
        <ul
          class="pages-nav__list pages-nav__list--main-menu"
        >
          <li
            class="flex"
            data-testid="main-nav-list-item"
          >
            <a
              class="pages-nav__link pages-nav__link--main-menu"
              href="/characters"
            >
              [Translation for https://example.com/global/translations.ttl#charactersPageTitle does not exist]
            </a>
          </li>
          <li
            class="flex"
            data-testid="main-nav-list-item"
          >
            <a
              class="pages-nav__link pages-nav__link--main-menu pages-nav__link--selected pages-nav__link--child-selected"
              href="/rules"
            >
              [Translation for https://example.com/global/translations.ttl#rulesPageTitle does not exist]
            </a>
            <button
              class="pages-nav__button pages-nav__button--main-menu lg:hidden"
              data-testid="nav-item-sub-menu-button"
              type="button"
            >
              <svg
                class="icon"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 6h16M4 12h16M4 18h16"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </li>
        </ul>
      </nav>
    </header>
    <button
      class="bg-black fixed top-0 left-0 w-screen h-screen opacity-80 z-0 lg:hidden hidden"
      data-testid="layout-fade"
      type="button"
    >
      Â 
    </button>
    <div
      class="main-container main-container--content main-container--main flex-1 flex flex-col lg:flex-row lg:space-x-2"
    >
      <div
        class="layout__sub-menu hidden lg:block"
        data-testid="layout-sub-menu"
      >
        <nav
          class="pages-nav pages-nav--sub-menu"
        >
          <button
            class="button lg:hidden bg-transparent border-transparent text-right px-2 w-full hover:bg-gray-900 hover:text-white dark:text-white focus:ring-gray-900 mb-9"
            data-testid="sub-menu-nav-close-button"
            type="button"
          >
            [Translation for https://example.com/global/translations.ttl#close does not exist]Â 
            <svg
              class="icon"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 18L18 6M6 6l12 12"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
          <ul
            class="nav-pages__list nav-pages__list--sub-menu"
          >
            <li
              class="flex"
            >
              <a
                class="pages-nav__link pages-nav__link--sub-menu pages-nav__link--selected pages-nav__link--child-selected"
                href="/rules"
              >
                [Translation for https://example.com/global/translations.ttl#rulesPageTitle does not exist]
              </a>
            </li>
            <li
              class="flex"
            >
              <a
                class="pages-nav__link pages-nav__link--sub-menu"
                href="/rules/classes"
              >
                [Translation for https://example.com/global/translations.ttl#classesPageTitle does not exist]
              </a>
            </li>
            <li
              class="flex"
            >
              <a
                class="pages-nav__link pages-nav__link--sub-menu"
                href="/rules/races"
              >
                [Translation for https://example.com/global/translations.ttl#racesPageTitle does not exist]
              </a>
            </li>
            <li
              class="flex"
            >
              <a
                class="pages-nav__link pages-nav__link--sub-menu"
                href="/rules/equipment"
              >
                [Translation for https://example.com/global/translations.ttl#equipmentPageTitle does not exist]
              </a>
            </li>
            <li
              class="flex"
            >
              <a
                class="pages-nav__link pages-nav__link--sub-menu"
                href="/rules/spells"
              >
                [Translation for https://example.com/global/translations.ttl#spellsPageTitle does not exist]
              </a>
            </li>
            <li
              class="flex"
            >
              <a
                class="pages-nav__link pages-nav__link--sub-menu"
                href="/rules/monsters"
              >
                [Translation for https://example.com/global/translations.ttl#monstersPageTitle does not exist]
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        class="flex-1 lg:flex-auto flex flex-col max-w-prose min-h-full"
      >
        <main
          class="flex-1"
        >
          test
        </main>
        <footer
          class="main-container main-container--content mt-8 text-center sm:text-left xs:flex xs:space-x-4 items-center justify-center sm:justify-start"
        >
          <div
            class="md:flex-grow pb-2 flex space-x-1 justify-center sm:justify-start"
          >
            <a
              class="link"
              href="/about"
            >
              [Translation for https://example.com/global/translations.ttl#aboutTitle does not exist]
            </a>
            <a
              class="link"
              href="/faq"
            >
              [Translation for https://example.com/global/translations.ttl#faqShort does not exist]
            </a>
          </div>
          <div
            class="pb-2 flex xs:space-x-2 space-y-2 xs:space-y-0 flex-col text-center xs:flex-row xs:justify-center"
          >
            <button
              class="button"
              data-testid="dark-mode-selector-button"
              type="button"
            >
              <span>
                ðŸŒ™Â [Translation for https://example.com/global/translations.ttl#darkModeTurnOn does not exist]
              </span>
            </button>
          </div>
        </footer>
      </div>
      <div
        class="hidden lg:block lg:flex-1"
      />
    </div>
  </div>
</DocumentFragment>
`;
